classes/class-wc-connect-shipping-label.php

```diff
@@ -353,19 +353,27 @@ protected function get_form_data( WC_Order $order ) {
 
 		protected function get_states_map() {
 			$result = array();
-			foreach ( WC()->countries->get_countries() as $code => $name ) {
-				$result[ $code ] = array( 'name' => html_entity_decode( $name ) );
-			}
-			foreach ( WC()->countries->get_states() as $country => $states ) {
-				$result[ $country ]['states'] = array();
-				foreach ( $states as $code => $name ) {
-					if ( 'US' === $country && in_array( $code, $this->unsupported_states ) ) {
-						continue;
+			$all_countries = WC()->countries->get_countries();
+			$all_states = WC()->countries->get_states();
```

Comment:
Instead of getting the global states array here, what do you think about using `WC()->countries->get_states( $country_code )` instead of `$states = $all_states[ $code ];` below to get the list of states for each allowed country?